### PR TITLE
#49 save survey answers

### LINKED ISSUE #49: Save survey responses

### PR DESCRIPTION
**Pull request by MichaelDysart**
Close #49  
Close #51

### CODE REVIEW

File: src/main/java/webApp/webController.java [hunk 1]
```diff
@@ -25,43 +30,120 @@ public String index(Model model){
...
-            } else if(question.getType().equals("numberQuestion")) {
-                questionList.add(new NumberQuestion(question.getQuestion(), question.getMin(), question.getMax()));
+        ArrayList<String> questionTexts = new ArrayList<>();
+
+        if (surveyMessage.getQuestions() == null) {
+            return new Response(null, "error", "Questions were null");
+        }
+
+        for (QuestionMessage questionMsg : surveyMessage.getQuestions()) {
+            if (questionMsg == null) {
```
Comment (JacksonSchoe) [inline src/main/java/webApp/webController.java:43]: Good job implementing error handling.

File: src/main/java/webApp/webController.java [hunk 2]
```diff
@@ -25,43 +30,120 @@ public String index(Model model){
...
-                questionList.add(new OpenEndedQuestion(question.getQuestion()));
-            } else if(question.getType().equals("numberQuestion")) {
-                questionList.add(new NumberQuestion(question.getQuestion(), question.getMin(), question.getMax()));
+        ArrayList<String> questionTexts = new ArrayList<>();
+
+        if (surveyMessage.getQuestions() == null) {
+            return new Response(null, "error", "Questions were null");
+        }
+
+        for (QuestionMessage questionMsg : surveyMessage.getQuestions()) {
```
Comment (JacksonSchoe) [inline src/main/java/webApp/webController.java:42]: Maybe more comments here explaining what is going

File: src/main/java/webApp/webController.java [hunk 3]
```diff
@@ -25,43 +30,120 @@ public String index(Model model){
...
 
         Collection<QuestionMessage> questionMessages = new ArrayList<>();
 
-        Collection<Survey> surveys = repo.findByName(name);
-        if (surveys.size() == 1 ) {
+        Optional<Survey> survey = repo.findById(id);
+        if (survey.isPresent()) {
 
-            for (Question question : surveys.iterator().next().getQuestions()) {
+            for (Question question : survey.get().getQuestions()) {
```
Comment (JacksonSchoe) [inline src/main/java/webApp/webController.java:80]: same with here

File: src/main/java/webApp/webController.java [hunk 4]
```diff
@@ -25,43 +30,120 @@ public String index(Model model){
...
 
-            return new SurveyMessage(name, questionMessages);
+            return new SurveyMessage(survey.get().getId(), "ok", survey.get().getName(), questionMessages);
+        }
+        return new SurveyMessage(null, "error", "", questionMessages);
+    }
+
+    @PostMapping(value = "/addAnswers", consumes = "application/json", produces = "application/json")
+    @ResponseBody
+    public Response addAnswers(@RequestBody SurveyMessage surveyMessage) {
```
Comment (JacksonSchoe) [inline src/main/java/webApp/webController.java:97]: And more comments here

**Review by JacksonSchoe — APPROVED** [review]
Everything is fine, I would like more comments around, but not entirely necessary. The overall functionality is good.

File: src/main/java/survey/Survey.java [hunk 5]
```diff
@@ -39,35 +40,36 @@ public Survey(String name, Collection<Question> questions){
...
     /*
-     * Sets the name
-     * @param name {String}
+     * Sets the id
+     * @param id {int}
      */
-    public void setName(String name) {
-        this.name = name;
+    public void setId(Integer id) {
+        this.id = id;
```
Comment (Fahid-Mannan) [inline src/main/java/survey/Survey.java:47]: Weren't these modified in a different PR?

**Review by Fahid-Mannan — COMMENTED** [review]
Did u use the same branch for multiple issues? It looks like a majority of this code is from another PR. Also please add a description the PR in addition to the GitHub issue number

File: src/main/java/survey/Survey.java (hunk 5 above)
Comment (MichaelDysart) [reply src/main/java/survey/Survey.java:47]: Yes. The PRs build on each other.

**Review by rinadina — APPROVED** [review]

File: src/main/java/survey/Survey.java (hunk 5 above)
Comment (MichaelDysart) [reply src/main/java/survey/Survey.java:47]: They need to be merged in the correct order. Look at them by order of date.

**Pull request merged**